src/Language/PureScript/Make/Actions.hs

```diff
@@ -345,55 +327,61 @@ data ForeignModuleType = ESModule | CJSModule deriving (Show)
 
 -- | Check that the declarations in a given PureScript module match with those
 -- in its corresponding foreign module.
-checkForeignDecls :: CF.Module ann -> FilePath -> Make (ForeignModuleType, S.Set Ident)
+checkForeignDecls :: CF.Module ann -> FilePath -> Make (Either MultipleErrors (ForeignModuleType, S.Set Ident))
 checkForeignDecls m path = do
   jsStr <- T.unpack <$> readTextFile path
-  js <- either (errorParsingModule . Bundle.UnableToParseModule) pure $ JS.parseModule jsStr path
-
-  (foreignModuleType, foreignIdentsStrs) <-
-    case (,) <$> getForeignModuleExports js <*> getForeignModuleImports js of
-      Left reason -> errorParsingModule reason
-      Right (Bundle.ForeignModuleExports{..}, Bundle.ForeignModuleImports{..})
-        | not (null cjsExports && null cjsImports)
-        , null esExports
-        , null esImports -> do
-            let deprecatedFFI = filter (elem '\'') cjsExports
-            unless (null deprecatedFFI) $
-              errorDeprecatedForeignPrimes deprecatedFFI
-
-            pure (CJSModule, cjsExports)
-        | otherwise -> do
-            unless (null cjsImports) $
-              errorUnsupportedFFICommonJSImports cjsImports
-
-            unless (null cjsExports) $
-              errorUnsupportedFFICommonJSExports cjsExports
-
-            pure (ESModule, esExports)
-
-  foreignIdents <- either
-                     errorInvalidForeignIdentifiers
-                     (pure . S.fromList)
-                     (parseIdents foreignIdentsStrs)
-  let importedIdents = S.fromList (CF.moduleForeign m)
-
-  let unusedFFI = foreignIdents S.\\ importedIdents
-  unless (null unusedFFI) $
-    tell . errorMessage' modSS . UnusedFFIImplementations mname $
-      S.toList unusedFFI
-
-  let missingFFI = importedIdents S.\\ foreignIdents
-  unless (null missingFFI) $
-    throwError . errorMessage' modSS . MissingFFIImplementations mname $
-      S.toList missingFFI
-
-  pure (foreignModuleType, foreignIdents)
+
+  let
+    parseResult :: Either MultipleErrors JS.JSAST
+    parseResult = first (errorParsingModule . Bundle.UnableToParseModule) $ JS.parseModule jsStr path
+  either (pure . Left) (fmap Right . checkFFI) parseResult
+
   where
   mname = CF.moduleName m
   modSS = CF.moduleSourceSpan m
 
-  errorParsingModule :: Bundle.ErrorMessage -> Make a
-  errorParsingModule = throwError . errorMessage' modSS . ErrorParsingFFIModule path . Just
+  checkFFI :: JS.JSAST -> Make (ForeignModuleType, S.Set Ident)
+  checkFFI js = do 
+    (foreignModuleType, foreignIdentsStrs) <-
+        case (,) <$> getForeignModuleExports js <*> getForeignModuleImports js of
+          Left reason -> throwError $ errorParsingModule reason
```

Comment:
Looks like this [didn't used to have a `throwError` in front of it](https://github.com/purescript/purescript/pull/4250/files#diff-889762f9d56ac6e7b5dcb3db0b2fca495526bde7fd588e07fdcacaa7d4aca1feL355). The `throwError` was moved into `errorParsingModule`. Would you mind undoing that? I think this change isn't needed for this PR, but correct me if I'm wrong.

Reply:
Gotcha!

Reply:
Hmm so the reason is that I'm using `errorParsingModule` also in line [336](https://github.com/purescript/purescript/pull/4250/files#diff-889762f9d56ac6e7b5dcb3db0b2fca495526bde7fd588e07fdcacaa7d4aca1feR336). I could keep it as I was but then I would need to handle that line differently, don't know if that makes much sense

Reply:
Ah, ok. Nevermind then.